Larisa/quizz manager

LarisaYeremyan opened the pull request.

ntermartirosyan committed:
style(eslint): add PascalCase support to method naming conventions

ntermartirosyan committed:
feat(user_management_v2): add E2E tests for user management with pagination

- Introduce comprehensive Cypress E2E test for seeding data and verifying pagination functionality.
- Add `UserManagementPageV2` utility methods for interacting with UI components.
- Enhance `UserManagementEndpoints` to support seeding user data via new endpoint.
- Update `UserManagementBuilders` with `seedData` method for test data setup.

ntermartirosyan committed:
style(user_management_v2): fix inconsistent formatting in tests and utilities

- Standardize code formatting across `UserManagementBuilders`, `UserManagementPageV2`, and E2E test files.
- Resolve inconsistent indentation, spacing, and punctuation for improved readability and maintainability.

narekntm committed:
Merge pull request #16 from narekntm/UserManagement_v2

User management v2

ntermartirosyan committed:
feat(user_management): add generators for seed data and improve reusable logic

- Introduce `UserManagementGenerators` for generating default and random user data.
- Update `UserManagementBuilders` with `AdminLogin` method and dynamic seed data generation.
- Simplify test setup by leveraging `UserManagementGenerators` for test scenarios.
- Remove unused `Promotions` subscription type from models.
- Refactor tests to use reusable logic for enhanced maintainability.

narekntm committed:
Merge pull request #24 from narekntm/UserManagement_v2

feat(user_management): add generators for seed data and improve reusa…

ntermartirosyan committed:
feat(quiz_manager): introduce admin and user interfaces for quiz management

- Add `admin.html` and `user.html` for managing and interacting with quizzes.
- Implement dynamic quiz creation, listing, and user submissions via `admin.js` and `user.js`.
- Add `quiz-view.html` and `quiz-view.js` for viewing and submitting quizzes.
- Include reusable `api.js` module for handling API requests with error handling.
- Introduce modular CSS (`styles.css`) for consistent and responsive UI styling.
- Update `package.json` with scripts to serve the `quiz_manager` frontend and backend.
- Add and configure new dependencies like `cookie-parser` and upgraded `uuid`.

ntermartirosyan committed:
feat(quiz_manager): enhance form validation and improve UI/UX consistency

- Add `novalidate` to form tags for client-side validation control.
- Introduce form validation feedback with visual styles for invalid inputs.
- Refactor quiz creation and login handlers to leverage browser validation.
- Reorganize and optimize CSS for buttons, forms, and disabled states.
- Update API endpoints (`POST` to `PATCH`) for quiz state updates and streamline response handling.
- Improve error handling in backend for unauthorized access and duplicate submissions.
- Standardize frontend code structure and fix minor inconsistencies in scripts.

ntermartirosyan committed:
feat(admin): refactor and modularize admin script for quiz management

- Break `admin.js` into smaller, reusable functions for enhanced modularity.
- Add `initAdminPage` for streamlined initialization and readability.
- Introduce custom assignment feature with dynamic user selection.
- Improve validation logic to enforce non-empty quiz titles and questions.
- Enhance quiz form reset behavior to clean up questions and assignment options.
- Add `/api/users` endpoint in the backend to fetch non-admin users.

ntermartirosyan committed:
feat(quiz_manager): improve UI components and enhance API usability

- Add `apiPatch` method to `api.js` for more precise HTTP method usage.
- Refactor admin actions to use `PATCH` requests for publishing and archiving quizzes.
- Enhance quiz listing UI with `status-badges` and action buttons for better clarity.
- Improve submission view UI with expandable sections and total count display.
- Add global expand/collapse toggle for submission details in `view-submissions.js`.
- Update server.js to add sample quizzes and standardize coding style.
- Introduce CSS updates for submission entries, badges, and quiz headers.

ntermartirosyan committed:
style(css): reorganize and optimize `styles.css` for clarity and maintainability

- Refactor CSS structure with clear sections for layout, forms, buttons, and responsiveness.
- Remove redundant styles and consolidate duplicate rules.
- Introduce new styles for headers, containers, and improved UI consistency.
- Add responsive design improvements for better mobile compatibility.

narekntm committed:
Merge pull request #27 from narekntm/quizzManager

Quizz manager

ntermartirosyan committed:
style(html): adjust indentation and update document type for consistency

- Normalize indentation across all HTML files for improved readability.
- Change `<!DOCTYPE html>` to lowercase `<!doctype html>` for better standards compliance.
- Ensure no behavioral changes, focusing solely on formatting improvements.

narekntm committed:
Merge pull request #28 from narekntm/quizzManager

style(html): adjust indentation and update document type for consistency

LarisaYeremyan committed:
Merge branch 'main' of https://github.com/narekntm/cybdg into LarisaYeremyan

ntermartirosyan committed:
feat(quiz_manager): update API config, scripts, and proxy setup for improved development workflow

- Change `baseUrl` in `api.js` to align frontend and backend URLs consistently.
- Add and configure `proxy-server.js` for local development with frontend/backend proxying.
- Update `serve` scripts in `package.json` to use consistent ports and include a combined runner.
- Introduce `config/quizzLecture.config.ts` for Cypress E2E testing with updated configuration.
- Add `concurrently` and `http-proxy-middleware` dependencies for streamlined development setup.
- Adjust backend server port to match new proxy setup requirements.

ntermartirosyan committed:
feat(quiz_manager): modularize CSS and add seed data for better structure and testing

- Split `styles.css` into modular CSS files for buttons, forms, quizzes, and other components.
- Update `admin.html` and other HTML files to reference the new modular CSS files.
- Add `seed.js` to generate test users, quizzes, and submissions for easier development and testing.
- Refactor `server.js` to use `seed.js` for data initialization.
- Organize backend code for improved maintainability and reusability.

ntermartirosyan committed:
feat(quiz_manager): enhance dynamic quiz generation and UI flexibility

- Add `generateOptions` utility in `seed.js` for dynamic option generation, improving test data variety.
- Update dummy quizzes to include randomized options for better testing coverage.
- Refactor quiz rendering logic to support enhanced UI flexibility and improve maintainability.
- Introduce CSS updates to improve quiz form styling, responsiveness, and accessibility.
- Standardize backend code formatting and logging for consistent style.

ntermartirosyan committed:
feat(quiz_manager): enhance submission UI/UX with interactive elements

- Improve submission cards with hover effects, transitions, and expandable details.
- Add global expand/collapse button for streamlined navigation in `view-submissions.js`.
- Enhance CSS for better responsiveness, readability, and consistent styling.
- Refactor JavaScript for cleaner event handling and improved code maintainability.

ntermartirosyan committed:
feat(quiz_manager): refine UI styles and enhance form/quiz behavior

- Update color scheme for buttons, headers, and badges for improved visual consistency.
- Introduce minimal CSS reset for standardized styling across browsers.
- Enhance quiz creation form with labeled inputs, flexible layout, and custom assignment dropdown.
- Add hover effects and transitions for buttons and badges to improve interactivity.
- Update `admin.html` structure for better accessibility and layout clarity.
- Refactor JavaScript for improved user assignment and modular handling of dynamic elements.

ntermartirosyan committed:
feat(quiz_manager): enhance UI, form behavior, and modular structure

- Add `user.css` for dedicated user quiz/submission styling.
- Introduce toggles for quiz creation form in `admin.html` for better UI flexibility.
- Improve quiz and submission layout in user and admin pages with updated CSS for responsiveness.
- Update JavaScript to include quiz count display and toggle behavior for better interactivity.
- Refactor `user.js` and `quiz-view.js` for consistent coding style and handling of quiz/submission data.
- Remove redundant styles and adjust margins in CSS files for cleaner layout.

ntermartirosyan committed:
feat(quiz_manager): enhance dynamic option handling and improve UI/UX

- Update quiz form to dynamically manage options based on question type.
- Add interactive elements for quiz creation, including add/remove option functionality.
- Display quiz and submission counts dynamically on the user dashboard.
- Improve CSS for grid-based layouts and enhance layout responsiveness for quizzes/submissions.
- Refactor JavaScript for better modularity, maintainability, and user interactions.

ntermartirosyan committed:
feat(quiz_manager): update terminology and enhance test mode features

- Standardize branding by updating "Quiz" to "Quizz" across HTML files and UI components.
- Add test-mode APIs for streamlined test user authentication and creation (`/api/test/auth` and `/api/test/users`).
- Enhance server seeding logic with fallback initialization of default users and quizzes.
- Improve button visibility by dynamically showing publish/archive actions based on quiz state.
- Minor UI adjustments for administrative dashboard with real-time admin username display.

ntermartirosyan committed:
feat(quiz_manager): rename admin role to manager and update related functionality

- Replace "admin" role with "manager" across frontend, backend, and seed data.
- Refactor role-based logic in authentication, authorization, and user filtering APIs.
- Update file names, CSS selectors, and HTML elements to reflect "manager" terminology.
- Modify seed data to initialize "manager" user and update quiz ownership associations.
- Ensure all references to "admin" are replaced with "manager" for consistency.

ntermartirosyan committed:
feat(quiz_manager): integrate toast notifications for improved feedback

- Add reusable `showToast` function in `toast.js` with customizable message types and durations.
- Implement toast notifications across key UI actions: logout, quiz form validation, quiz submissions, and submission loading.
- Update all HTML files to include `toast.css` and a `toast-container` element.
- Refactor error displays in JavaScript to use `showToast` instead of alert dialogs for consistency.
- Enhance user feedback with detailed error and success messages in a non-intrusive manner.
- Introduce `toast.css` for styling toast notifications with fade-in and fade-out animations.

ntermartirosyan committed:
feat(quiz_manager): improve quiz assignment handling and UI feedback

- Update `assignedUsers` logic across backend and frontend to standardize as arrays.
- Modify all relevant APIs, seed data, and form submission handling to reflect updated structure.
- Enhance user feedback with toast notifications for login errors and quiz actions.
- Improve CSS for quiz manager list with clearer user and description sections.
- Add dynamic rendering of quiz assignees with truncated display for large lists.
- Refactor proxy and backend log messages for clarity and consistency.

ntermartirosyan committed:
style(css): fix indentation for consistency across modular CSS files

- Adjust indentation in all CSS files for improved readability and maintainability.
- Ensure consistent formatting across buttons, forms, manager, and quiz-related styles.

LarisaYeremyan committed:
Quizz Management

LarisaYeremyan committed:
some changes to quizz

LarisaYeremyan committed:
some changes after eslint

ntermartirosyan commented on Cypress/Fixtures/Pages/Larisa/QuizzManagementPage.ts:
```diff
@@ -0,0 +1,9 @@
+export class QuizzManagementLoginPage {
+  static title = () => cy.get(".login-container h1");
+  static emailLbl = () => cy.get("#email").prev("label");
+  static emailInput = () => cy.get("input#email");
+  static passwordLbl = () => cy.get("#password").prev("label");
+  static passwordInput = () => cy.get("#password");
+  static submitBtn = () => cy.get('button[type="submit"]');
+  static errorMessage = () => cy.get("#error-message");
+}
```
File name should be the same as class name inside it

LarisaYeremyan committed:
Quizz changes after pull request

LarisaYeremyan committed:
quizz

ntermartirosyan committed:
feat(quiz_manager): enhance quiz and submission handling with sorting and metadata

- Randomize quizzes display in `manager.js` and `user.js` for better variety.
- Add sorting of submissions by creation date in `view-submissions.js` and `user.js` for improved clarity.
- Include metadata (e.g., dataset IDs) in dynamically generated list items to support future expandability.

LarisaYeremyan committed:
quizz

narekntm committed:
Merge pull request #32 from narekntm/quizzManager

Quizz manager

LarisaYeremyan committed:
Merge branch 'main' of https://github.com/narekntm/cybdg into larisa/QuizzManager
